task/k8s/resources/data_source_persistent_volume.go

```diff
@@ -0,0 +1,52 @@
+package resources
+
+import (
+	"context"
+
+	kubernetes_core "k8s.io/api/core/v1"
+	kubernetes_errors "k8s.io/apimachinery/pkg/api/errors"
+	kubernetes_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"terraform-provider-iterative/task/common"
+	"terraform-provider-iterative/task/k8s/client"
+)
+
+// NewExistingPersistentVolumeClaim creates a new ExistingPersistentVolumeClaim object.
+func NewExistingPersistentVolumeClaim(client *client.Client, storageParams common.RemoteStorage) *ExistingPersistentVolumeClaim {
+	return &ExistingPersistentVolumeClaim{
+		client: client,
+		params: storageParams,
+	}
+}
+
+// ExistingPersistentVolumeClaim refers to a pre-allocated persistent volume to be used
+// as storage for the job.
+type ExistingPersistentVolumeClaim struct {
+	client   *client.Client
+	params   common.RemoteStorage
+	resource *kubernetes_core.PersistentVolumeClaim
+}
+
+// Read verifies the persistent volume.
+func (p *ExistingPersistentVolumeClaim) Read(ctx context.Context) error {
+	persistentVolumeClaim, err := p.client.Services.Core.PersistentVolumeClaims(p.client.Namespace).Get(ctx, p.params.Container, kubernetes_meta.GetOptions{})
+	if err != nil {
+		if statusErr, ok := err.(*kubernetes_errors.StatusError); ok && statusErr.ErrStatus.Code == 404 {
+			return common.NotFoundError
+		}
+		return err
+	}
+
+	p.resource = persistentVolumeClaim
+	return nil
+}
+
+// VolumeInfo returns information for attaching the persistent volume claim to the job.
+func (p *ExistingPersistentVolumeClaim) VolumeInfo(ctx context.Context) (string /*subpath*/, *kubernetes_core.PersistentVolumeClaimVolumeSource) {
+	pvc := &kubernetes_core.PersistentVolumeClaimVolumeSource{
+		ClaimName: p.params.Container,
+	}
+	return p.params.Path, pvc
+
+}
```

Comment:
Why use a function plus an interface instead of simply exposing the values as part of the `ExistingPersistentVolumeClaim` structure? 🤔

Reply:
Because we have two implementations for that interface - `PersistentVolumeClaim` and `ExistingPersistentVolumeClaim`

Reply:
Why not just use a separate data source like in the other backends? 🤔

Reply:
It is a data source. This is the exact same pattern used in other backends. We're just using the interface to pass to the `job` resource.